pkg/services/internal/whatsonchain/service.go

```diff
@@ -39,10 +42,12 @@ func New(httpClient *resty.Client, logger *slog.Logger, apiKey string, network d
 		SetRetryMaxWaitTime(Retries * RetriesWaitTime)
 
 	return &WhatsOnChain{
-		httpClient: client,
-		apiKey:     apiKey,
-		url:        fmt.Sprintf("https://api.whatsonchain.com/v1/bsv/%s", network),
-		logger:     logger,
+		httpClient:        client,
+		apiKey:            config.ApiKey,
+		url:               fmt.Sprintf("https://api.whatsonchain.com/v1/bsv/%s", network),
+		logger:            logger,
+		bsvExchangeRate:   config.BsvExchangeRate,
```

Comment:
I would initialize this field with `config.BSVExchangeRate{}` this way, you won't need to check if it's nil and the timestamp will be year 1 so it's always outdated.